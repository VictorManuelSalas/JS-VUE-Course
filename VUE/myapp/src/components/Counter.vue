<template>
    <div id="counter">
        <h2 class="title">{{title}}</h2>
        <p data-testid="counter">{{ num }}</p>
        <label for="elevacion">Ingresa el numero de elevacion </label>
        <input id="elevacion" type="text" v-model="eleva">
        <p>{{ num }} <sup>{{ eleva }}</sup> = {{ num * eleva }}</p>

        <button @click="aumentar">Ingrementar numero base</button>

        <Result :result="num * eleva" />
    </div>
</template>

<script>
import { ref } from 'vue';
import Result from '@/components/Result.vue'
export default {
    name: 'Counter',
    components: {
        Result,
    },
    data() {
        return {
            num: 0,
            eleva: 0
        }
    },
    methods: {
        aumentar() {
            this.num = ref(this.num)
            this.num = this.num + 1
        }
    },
    props: {
        test_propsts: {
            typeof: Number,
            default: 0
        },
        title: String,
        default: 'Counter'
    }
}
</script>

<style></style>