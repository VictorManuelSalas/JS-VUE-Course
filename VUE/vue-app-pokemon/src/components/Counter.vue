<template>
    <div>
        Puntuacion: {{counter}}
    </div>
</template>

<script>
export default {
    props:{
        counter:{
            typeof: Number,
            default: 0
        }
    },
    data() {
        return {
            
        }
    }
}
</script>

<style scoped>
div {
    background-color: rgb(255, 255, 255);
    color: rgb(0, 0, 0);
    border-radius: 2rem;
    padding: 1rem 2rem;
    margin: 1rem 0;
}
</style>