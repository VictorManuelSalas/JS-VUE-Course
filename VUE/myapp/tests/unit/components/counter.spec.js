//------------------------------------Evaluar cambios en el codigo-------------------------------------------------//
//ShallowMount hace montar lijeramente
import { shallowMount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter Component", () => {
  //Snapchot permite montar un componente de Vue de forma aislada para su prueba, sin tener en cuenta los componentes secundarios.
  test("Debe de hacer match con el snapshot", () => {

    //Se monta el componente y se almacena en una variable
    const wrapper = shallowMount(Counter);

    //Se le extrae el html al componente montado en wrapper
    //Con toMatchSnapshot al ejecutarse se creara una carpeta de snapshot y un archivo donde se almacena el html para asi ver las comparaciones
    expect(wrapper.html()).toMatchSnapshot()
  });
});


//Esto arrojara una respuesta, si el codigo permanece igual saldra como Passed, pero si se modifico algo en el html 
//arrojara error y mostrara donde se modifico para asi si no es algo que querramos, vamos y lo removemos.

//Pero si es un cambio necesario, se debe de actualizar la prueba del match para que almacene el nyevo codigo en el snapshot mediante -> npm run test:unit -u  
//Antes de eso deben de agregar en el package.json bajo el -> "test:unit": "vue-cli-service test:unit" el siguiente comando ➡️ "test:unit -u": "vue-cli-service test:unit -u"
