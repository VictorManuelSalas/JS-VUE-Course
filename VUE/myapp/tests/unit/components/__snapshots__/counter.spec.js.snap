// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe de hacer match con el snapshot 1`] = `
<div id="counter">
  <h2 class="title">Counter</h2><label for="elevacion">Ingresa el numero de elevacion </label><input id="elevacion" type="text">
  <p>0 <sup>0</sup> = 0</p><button>Ingrementar numero base</button>
  <result-stub result="0"></result-stub>
</div>
`;
